curriculum/challenges/english/10-coding-interview-prep/data-structures/create-a-hash-table.md

```diff
@@ -16,7 +16,7 @@ Here, we won't be concerned with the details of hashing or hash table implementa
 
 # --instructions--
 
-Let's create the basic functionality of a hash table. We've created a naive hashing function for you to use. You can pass a string value to the function `hash` and it will return a hashed value you can use as a key for storage. Store items based on this hashed value in the `this.collection` object. Create these three methods: `add`, `remove`, and `lookup`. The first should accept a key value pair to add to the hash table. The second should remove a key-value pair when passed a key. The third should accept a key and return the associated value or `null` if the key is not present.
+Let's create the basic functionality of a hash table. We've created a naive hashing function for you to use. You can pass a string value to the function `hash` and it will return a hashed value you can use as a key for storage. Store items based on this hashed value in the `this.collection` object. Create these three methods in the following order: `add`, `lookup`, and `remove`. The first should accept a key value pair to add to the hash table. The second should accept a key and return the associated value or `null` if the key is not present. The third should remove a key-value pair when passed a key.
```

Comment:
> Create these three methods in the following order: `add`, `lookup`, and `remove`.

Why are we instructing the order here?  It should not make any difference what order I write a method.

Reply:
@RandellDawson The code in the issue is a correct implementation of a hash map, but we still fail it. We either have to enforce the data structure of `this.collection` to be able to test it or use the `add` / `lookup` function that the user created (in which case, they have to write them before the remove function). I suppose if you were actually writing a hash map from scratch, you'd need to write those functions in that order

Reply:
We could also change the order of the tests to reinforce some methods are written first I suppose.  I still say the order in which methods appear in a function/class should not matter.

Reply:
I'd say it's pretty safe to assume `remove` method needs working `add` method. However in the basic test of `remove`, if there isn't already written `add` by user, `addMethodSolution` could be patched in, as test checks internal `collection`. Depending what kind implementation user has in mind it still might not work.

Part of this comes from the challenge assumptions
>Here, we won't be concerned with the details of hashing or hash table implementation, we will just try to get a general sense of how they work.

Reply:
Patching in the add when they haven't written one yet doesn't seem like a good idea, because it would make the tests more opaque. The remove method could pass, but when you write your own add method, it starts to fail. The user would have no idea why that would be happening. At least with telling the user that they need to write the functions in a particular order, they know why it's failing.

I don't think making them write it in a particular order is bad as long as we're really clear that you have to do so. I really don't like the idea of enforcing a certain data structure for collisions, because that takes some of the creativity out of the problem.

I'm open to a different way to approach this, but I don't see a way around this problem

Reply:
I think I would be fine with this. It does feel like `in the following order` could be a little misinterpreted. The order in the code doesn't matter, and the order that they are written doesn't really matter either. It's just that the `remove` tests won't pass until the `add` method is correct. I liked my original message a little better, which would be to put a note next to the test or in the instructions that `the remove test wont pass until the add works` or something like that. However, this PR does fix the issue - and I think it's probably good enough if it needs to be.

Reply:
Ah, I can see how that can be misinterpreted. I'll update the PR